caiman/utils/stats.py

```diff
@@ -181,24 +183,37 @@ def fnc(x):
     else:
         # Create the function that we can use for the half-sample mode
         err = True
-        while err:
+        err_count = 0
+        max_err_count = 10
+        while err==True and err_count < max_err_count:
```

Comment:
`while err and err_count`...

Testing boolean values for ==True is an antipattern